src/assets/HTMLAsset.js

```diff
@@ -160,8 +165,71 @@ class HTMLAsset extends Asset {
     }
   }
 
-  generate() {
-    return this.isAstDirty ? render(this.ast) : this.contents;
+  async generate() {
+    let parts = [];
+    this.ast.walk(node => {
+      if (node.tag === 'script' || node.tag === 'style') {
+        if (node.content) {
+          let value = node.content.join('').trim();
+          if (value) {
+            parts.push({
+              type:
+                node.tag === 'style'
+                  ? 'css'
+                  : node.attrs && node.attrs.type
+                    ? SCRIPT_TYPES[node.attrs.type] || 'js'
+                    : 'js',
+              value
+            });
+          }
+        }
+      }
+
+      if (node.attrs && node.attrs.style) {
+        parts.push({
+          type: 'css',
+          value: node.attrs.style
+        });
+      }
+
+      return node;
+    });
+
+    return parts;
+  }
+
+  async postProcess(generated) {
+    // Hacky way of filtering out the css hmr JS
+    // Related: https://github.com/parcel-bundler/parcel/issues/1389
+    generated = generated.filter(
```

Comment:
This line might cause issues, related to this issue: https://github.com/parcel-bundler/parcel/issues/1389 
There is currently no way to tell in generated you don't want the hmr code for css assets.